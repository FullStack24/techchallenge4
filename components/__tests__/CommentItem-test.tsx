import * as React from 'react';
import renderer from 'react-test-renderer';
import { IComment } from '@/interfaces/Comment';
import CommentItem from '../CommentItem';

it('deve renderizar o CommentItem corretamente', () => {
  const mockComment: IComment = {
    id: '1',
    content: 'This is a comment',
    author: 'John Doe',
    post_id: '1',
    created_at: new Date().toISOString(),
  };

  const tree = renderer.create(
    <CommentItem comment={mockComment} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
