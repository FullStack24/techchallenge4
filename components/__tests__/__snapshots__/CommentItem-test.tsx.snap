// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deve renderizar o CommentItem corretamente 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "elevation": 2,
      "marginBottom": 12,
      "padding": 12,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.1,
      "shadowRadius": 4,
    }
  }
>
  <Text
    style={
      {
        "color": "#333333",
        "fontWeight": "bold",
        "marginBottom": 4,
      }
    }
  >
    John Doe
  </Text>
  <Text
    style={
      {
        "color": "#555555",
      }
    }
  >
    This is a comment
  </Text>
</View>
`;
